opentelemetry-otlp/tests/smoke.rs

```diff
@@ -136,3 +151,150 @@ async fn smoke_tracer() {
     let first_event = first_span.events.first().unwrap();
     assert_eq!("my-test-event", first_event.name);
 }
+
+#[tokio::test(flavor = "multi_thread")]
```

Comment:
we now have automated integration tests. Could these tests be added there? It does not have anything validating the TLS code paths..
https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-otlp/tests/integration_test/tests

Reply:
+1., this needs to be moved to integrations tests.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/gen_test_cert.sh we can steal some scripts from here to generate the needed artifacts.